src/relay/pass/dead_code.cc

```diff
@@ -0,0 +1,98 @@
+/*!
+ * Copyright (c) 2018 by Contributors
+ *
+ * \file dead_code.cc
+ *
+ * \brief Remove code that does not effect the program result.
+ *
+ * The algorithm is implemented by two visitor:
+ * CalcDep turn an expr into a dependency graph of expr,
+ * GenLet turn the dependency graph into a let list, taking only the used value.
+ */
+#include <tvm/relay/pass.h>
+#include <tvm/relay/expr_functor.h>
+#include "./let_list.h"
+
+namespace tvm {
+namespace relay {
+
+bool IsBoolLit(const Expr & e, bool b) {
```

Comment:
Let us relax the check for now, allowing any integer type as long as the content is 1

Reply:
int64_t GetIntValue(size_t i) {
  if (dt == Int(32)) {
      return static_cast<const int64_t*>(c->data->data)[0]
  } else if (dt ==  Int(32)) {
      return static_cast<const int32_t*>(c->data->data)[0]
  } else {
     ...
  }
}